src/Plugins/SimplnxCore/src/SimplnxCore/Filters/Algorithms/ComputeFeatureBounds.cpp

```diff
@@ -0,0 +1,418 @@
+  float32 minZ = std::numeric_limits<float32>::max();
+
+  usize numComp = edges.getNumberOfComponents();
+  for(usize i = 0; i < edges.getNumberOfTuples(); i++)
+  {
+    if(featureIds[i] == activeFeature)
+    {
+      for(usize comp = 0; comp < numComp; comp++)
+      {
+        const IGeometry::SharedFaceList::value_type activeVertIndex = edges[(i * numComp) + comp];
+        float32 xVal = verts[(activeVertIndex * 3) + 0];
+        float32 yVal = verts[(activeVertIndex * 3) + 1];
+        float32 zVal = verts[(activeVertIndex * 3) + 2];
+
+        minX = std::min(minX, xVal);
+        minY = std::min(minY, yVal);
+        minZ = std::min(minZ, zVal);
+
+        maxX = std::max(maxX, xVal);
+        maxY = std::max(maxY, yVal);
+        maxZ = std::max(maxZ, zVal);
+      }
+    }
+  }
+
+  return std::make_pair(Point3Df(minX, minY, minZ), Point3Df(maxX, maxY, maxZ));
+}
+} // namespace Geometry1D
+namespace Geometry2D
+{
+std::pair<Point3Df, Point3Df> CalculateFeatureBounds(int32 activeFeature, const IGeometry::SharedVertexList::store_type& verts, const IGeometry::SharedFaceList::store_type& faces,
+                                                     const Int32AbstractDataStore& featureIds)
+{
+  float32 maxX = std::numeric_limits<float32>::lowest();
+  float32 maxY = std::numeric_limits<float32>::lowest();
+  float32 maxZ = std::numeric_limits<float32>::lowest();
+
+  float32 minX = std::numeric_limits<float32>::max();
+  float32 minY = std::numeric_limits<float32>::max();
+  float32 minZ = std::numeric_limits<float32>::max();
+
+  usize numComp = faces.getNumberOfComponents();
+  for(usize i = 0; i < faces.getNumberOfTuples(); i++)
+  {
+    if(featureIds[i] == activeFeature)
+    {
+      for(usize comp = 0; comp < numComp; comp++)
+      {
+        const IGeometry::SharedFaceList::value_type activeVertIndex = faces[(i * numComp) + comp];
+        float32 xVal = verts[(activeVertIndex * 3) + 0];
+        float32 yVal = verts[(activeVertIndex * 3) + 1];
+        float32 zVal = verts[(activeVertIndex * 3) + 2];
+
+        minX = std::min(minX, xVal);
+        minY = std::min(minY, yVal);
+        minZ = std::min(minZ, zVal);
+
+        maxX = std::max(maxX, xVal);
+        maxY = std::max(maxY, yVal);
+        maxZ = std::max(maxZ, zVal);
+      }
+    }
+  }
+
+  return std::make_pair(Point3Df(minX, minY, minZ), Point3Df(maxX, maxY, maxZ));
+}
+} // namespace Geometry2D
+
+template <bool ImageGeomV, bool Geom0DV, bool Geom1DV, bool Geom2DV>
+struct GeometryOptions
+{
+  static constexpr bool UsingImage = ImageGeomV;
+  static constexpr bool Using0D = Geom0DV;
+  static constexpr bool Using1D = Geom1DV;
+  static constexpr bool Using2D = Geom2DV;
+};
+
+using ImageOption = GeometryOptions<true, false, false, false>;
+using Geom0DOption = GeometryOptions<false, true, false, false>;
+using Geom1DOption = GeometryOptions<false, false, true, false>;
+using Geom2DOption = GeometryOptions<false, false, false, true>;
```

Comment:
I don't think you need this additional type to branch on the geometry type. The class already is templated around the geometry type itself and there's no case where you'd want more than 1 option active at a time.